libraries/standard/mqtt/src/mqtt.c

```diff
@@ -326,33 +322,41 @@ static int32_t sendPacket( MQTTContext_t * pContext,
     sendTime = pContext->callbacks.getTime();
 
     /* Loop until the entire packet is sent. */
-    while( bytesRemaining > 0UL )
+    while( ( bytesRemaining > 0UL ) && ( sendError == false ) )
     {
         bytesSent = pContext->transportInterface.send( pContext->transportInterface.pNetworkContext,
                                                        pIndex,
                                                        bytesRemaining );
 
-        if( bytesSent > 0 )
+        if( bytesSent <= 0 )
+        {
+            LogError( ( "Transport send failed. Error code=%d.", bytesSent ) );
+            totalBytesSent = bytesSent;
+            sendError = true;
+        }
+        else if( ( size_t ) bytesSent > bytesRemaining )
+        {
+            LogError( ( "Transport send returned more bytes than expected: "
+                        "BytesExpected=%lu, BytesReceived=%d",
+                        bytesRemaining,
+                        bytesSent ) );
+            sendError = true;
```

Comment:
What would the function return in this case?

Reply:
Let me leave the same comments as the recvExcat.